libpod/boltdb_state_internal.go

```diff
@@ -0,0 +1,193 @@
+package libpod
+
+import (
+	"encoding/json"
+
+	"github.com/boltdb/bolt"
+	"github.com/pkg/errors"
+)
+
+const (
+	idRegistryName    = "id-registry"
+	nameRegistryName  = "name-registry"
+	ctrConfigName     = "container-config"
+	ctrStateName      = "container-state"
+	netNSName         = "net-ns"
+	runtimeConfigName = "runtime-config"
+)
+
+var (
+	idRegistryBkt    = []byte(idRegistryName)
+	nameRegistryBkt  = []byte(nameRegistryName)
+	ctrConfigBkt     = []byte(ctrConfigName)
+	ctrStateBkt      = []byte(ctrStateName)
+	netNSBkt         = []byte(netNSName)
+	runtimeConfigBkt = []byte(runtimeConfigName)
+)
+
+// Check if the configuration of the database is compatible with the
+// configuration of the runtime opening it
+// If there is no runtime configuration loaded, load our own
+func checkRuntimeConfig(db *bolt.DB, runtime *Runtime) error {
+	var (
+		staticDir       = []byte("static-dir")
+		tmpDir          = []byte("tmp-dir")
+		runRoot         = []byte("run-root")
+		graphRoot       = []byte("graph-root")
+		graphDriverName = []byte("graph-driver-name")
+	)
+
+	err := db.Update(func(tx *bolt.Tx) error {
+		configBkt, err := getRuntimeConfigBucket()
+		if err != nil {
+			return err
+		}
+
+		err = validateDBAgainstConfig(configBkt, "static dir",
+			runtime.config.StaticDir, staticDir)
+		if err != nil {
+			return err
+		}
+
+		err = validateDBAgainstConfig(configBkt, "tmp dir",
+			runtime.config.TmpDir, tmpDir)
+		if err != nil {
+			return err
+		}
+
+		err = validateDBAgainstConfig(configBkt, "run root",
+			runtime.config.StorageConfig.RunRoot, runRoot)
+		if err != nil {
+			return err
+		}
+
+		err = validateDBAgainstConfig(configBkt, "graph root",
+			runtime.config.StorageConfig.GraphRoot, graphRoot)
+		if err != nil {
+			return err
+		}
+
+		err = validateDBAgainstConfig(configBkt, "graph driver name",
+			runtime.config.StorageConfig.GraphDriverName,
+			graphDriverName)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	})
+
+	return err
+}
+
+// Validate a configuration entry in the DB against current runtime config
+// If the given configuration key does not exist it will be created
+func validateDBAgainstConfig(bucket *bolt.Bucket, fieldName, runtimeValue string, keyName []byte) error {
+	keyBytes := bucket.Get(keyName)
+	if keyBytes == nil {
+		if err := bucket.Put(keyName, []byte(runtimeValue)); err != nil {
+			return errors.Wrapf(err, "error updating %s in DB runtime config", fieldName)
+		}
+	} else {
+		if runtimeValue != string(keyBytes) {
+			return errors.Wrapf(ErrDBBadConfig, "database %s %s does not match our %s %s",
```

Comment:
nit.  Just wondering if this error will make sense to the end user "database containername George does not match our containername Fred".  I know it's longer, but perhaps something like 

"validation error.  The value %s does not match %s for the field %s as expected." fieldname, string(keyBytes), fieldname

I'm open to different wording as you feel is warranted.